eva_submission/eload_ingestion.py

```diff
@@ -209,28 +210,37 @@ def check_variant_db(self):
-            for analysis_accession in analyses.values():
-                self.load_from_ena_from_project_or_analysis(analysis_accession)
-
-        else:
-            self.load_from_ena_from_project_or_analysis()
-
-    def load_from_ena_from_project_or_analysis(self, analysis_accession=None):
         """
-        Loads Analysis metadata from ENA into EVAPRO to the project associated with this ELOAD or to an analysis
-        if it is specified.
+        Loads Project and Analysis metadata from ENA into EVAPRO to the project associated with this ELOAD.
         """
-        # Current submission process never changes -c or -v
-        command = (f"perl {cfg['executable']['load_from_ena']} -p {self.project_accession} -c submitted -v 1 "
-                   f"-l {self._get_dir('scratch')} -e {str(self.eload_num)}")
-        if analysis_accession:
-            command += f' -A 1 -a {analysis_accession}'
+        loader = EvaProjectLoader()
+        sample_name_2_accession = self.eload_cfg.query('brokering', 'Biosamples', 'Samples')
         try:
-            command_utils.run_command_with_output('Load metadata from ENA to EVAPRO', command)
+            # Load entire project, or only analyses associated with this submission
+            if check_project_exists_in_evapro(self.project_accession):
+                analyses = self.eload_cfg.query('brokering', 'ena', 'ANALYSIS')
+                for analysis_accession in analyses.values():
+                    loader.load_project_from_ena(self.project_accession, self.eload_num, analysis_accession)
+            else:
+                loader.load_project_from_ena(self.project_accession, self.eload_num)
+
+            # Check aggregation type for each analysis in this submission to determine if we should load samples by file
+            # or by analysis
+            for analysis_alias in self.eload_cfg.query('validation', 'aggregation_check', 'analyses'):
+                aggregation_type = self.eload_cfg.query('validation', 'aggregation_check', 'analyses', analysis_alias)
+                if aggregation_type == 'basic':
+                    analysis_accession = self.eload_cfg.query('brokering', 'ena', 'ANALYSIS', analysis_alias)
+                    loader.load_samples_from_analysis(sample_name_2_accession, analysis_accession)
+                else:
+                    analysis_info = self.eload_cfg.query('brokering', 'analyses', analysis_alias)
+                    for vcf_file in analysis_info['vcf_files']:
+                        vcf_file_md5 = analysis_info['vcf_files'][vcf_file]['md5']
+                        loader.load_samples_from_vcf_file(sample_name_2_accession, vcf_file, vcf_file_md5)
+            loader.update_project_samples_temp1(self.project_accession)
+
+            self.refresh_study_browser()
             self.eload_cfg.set(self.config_section, 'ena_load', value='success')
-        except subprocess.CalledProcessError as e:
+        except Exception as e:
```

Comment:
The main difference  with the previous loader is that in case of an error we might have half loaded data since we're committing either the whole project or each analysis then the samples.
I don't see this as a major problem but we could end up with partially loaded data in EVAPRO.